gensim/test/test_doc2vec.py

```diff
@@ -123,6 +123,18 @@ def test_empty_errors(self):
         # input not empty, but rather completely filtered out
         self.assertRaises(RuntimeError, doc2vec.Doc2Vec, list_corpus, min_count=10000)
 
+    def test_oot_similarity(self):
+        """Test similarity of out of training sentences"""
+        empty_str = ''
+        rome_str = 'rome italy'
+        car_str = 'car'
+        corpus = list(DocsLeeCorpus(True))
+
+        model = doc2vec.Doc2Vec(min_count=1)
+        model.build_vocab(corpus)
+        self.assertEqual(int(model.docvecs.oot_similarity(model, '', '')), 1)
```

Comment:
This is not testing what you think it is. Repeated calls to inference with the exact same tokens don't necessarily result in identical vectors, because of randomness in the Doc2Vec algorithm. But it works here because these empty-strings result in no inference at all, which means each vector stays its initial untrained randomized value (which we have tried to make start in the same place for the same inputs). With actual tokens, you might not get a 1.0 similarity.